fondant/component.py

```diff
@@ -179,8 +179,14 @@ def _load_or_create_manifest(self) -> Manifest:
         return Manifest.from_file(self.args.input_manifest_path)
 
     @abstractmethod
-    def transform(self, *, dataframe: dd.DataFrame, **kwargs) -> dd.DataFrame:
-        """Abstract method for applying data transformations to the input dataframe."""
+    def transform(self, *args, **kwargs) -> dd.DataFrame:
```

Comment:
This led to typing issues before since the signature in the subclasses contain specific arguments instead of `**kwargs`. mypy has a [mechanism](https://github.com/python/mypy/issues/5876) to enable this, but it only works if you specify both `*args` and `**kwargs`.